src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java

```diff
@@ -534,14 +534,17 @@ SpawnResult execLocallyAndUpload(
         return result;
       }
     }
+
+    uploadLocalResults = uploadLocalResults &&
+        Status.SUCCESS.equals(result.status()) && result.exitCode() == 0;
     if (!uploadLocalResults) {
```

Comment:
In the `RemoteSpawnCache`, the equivalent condition was moved before the concurrent modification check. It probably doesn't matter, but it doesn't hurt to be consistent either way.